lib/core/tests/html_renderer.rs

```diff
@@ -18,57 +18,59 @@ extern "C" {
 
 #[cfg(test)]
 mod tests {
-    use basegl::system::web::dom::Scene;
-    use basegl::system::web::dom::Camera;
-    use basegl::system::web::dom::html::HTMLObject;
-    use basegl::system::web::dom::html::HTMLRenderer;
+    use basegl::display::camera::Camera2d;
+    use basegl::system::web::dom::html::HtmlScene;
+    use basegl::system::web::dom::html::HtmlObject;
+    use basegl::system::web::dom::html::HtmlRenderer;
     use basegl::system::web::StyleSetter;
     use basegl::system::web::get_performance;
     use web_test::*;
     use web_sys::Performance;
     use nalgebra::Vector3;
+    use logger::Logger;
 
     #[web_test(no_container)]
     fn invalid_container() {
-        let renderer = HTMLRenderer::new("nonexistent_id");
+        let renderer = HtmlRenderer::new("nonexistent_id");
         assert!(renderer.is_err(), "nonexistent_id should not exist");
     }
 
     #[web_test]
     fn object_behind_camera() {
-        let mut scene : Scene<HTMLObject> = Scene::new();
-        let renderer = HTMLRenderer::new("object_behind_camera")
+        let logger = Logger::new("object_behind_camera");
+
+        let mut scene = HtmlScene::new(logger.clone());
+        let renderer = HtmlRenderer::new("object_behind_camera")
                                     .expect("Renderer couldn't be created");
         assert_eq!(scene.len(), 0, "Scene should be empty");
 
         let view_dim = renderer.dimensions();
         assert_eq!((view_dim.x, view_dim.y), (320.0, 240.0));
 
-        let mut object = HTMLObject::new("div").unwrap();
+        let mut object = HtmlObject::new(logger.clone(), "div").unwrap();
         object.set_position(Vector3::new(0.0, 0.0, 0.0));
         object.dom.set_property_or_panic("background-color", "black");
         object.set_dimensions(100.0, 100.0);
-        scene.add(object);
+        scene.add_child(object);
 
-        let aspect_ratio = view_dim.x / view_dim.y;
-        let mut camera = Camera::perspective(45.0, aspect_ratio, 1.0, 1000.0);
+        let mut camera  = Camera2d::new(logger,view_dim.x,view_dim.y);
         // We move the Camera behind the object so we don't see it.
         camera.set_position(Vector3::new(0.0, 0.0, -100.0));
 
-        renderer.render(&mut camera, &scene);
+        renderer.render(&mut camera,&scene);
     }
 
-    fn create_scene(renderer : &HTMLRenderer) -> Scene<HTMLObject> {
-        let mut scene : Scene<HTMLObject> = Scene::new();
+    fn create_scene(logger:Logger, renderer:&HtmlRenderer) -> HtmlScene {
+        let mut scene: HtmlScene = HtmlScene::new(logger.clone());
```

Comment:
this is also clone on call side. The api should take reference to it instead (here and in other objects)

Reply:
Fixing